contracts/Versus.cdc

```diff
@@ -628,17 +720,38 @@ pub contract Versus {
 
     //initialize all the paths and create and link up the admin proxy
     init() {
-        self.totalDrops = (0 as UInt64)
 
         self.CollectionPublicPath= /public/versusCollection
+        self.CollectionPrivatePath= /private/versusCollection
         self.CollectionStoragePath= /storage/versusCollection
-        self.VersusAdminClientPublicPath= /public/versusAdminClient
-        self.VersusAdminClientStoragePath=/storage/versusAdminClient
-        self.VersusAdministratorStoragePath=/storage/versusAdmin
-        self.VersusAdministratorPrivatePath=/private/versusAdmin
+        self.VersusAdminClientPublicPath= /public/versusAdmin
+        self.VersusAdminClientStoragePath=/storage/versusAdmin
+
+
+        self.totalDrops = (0 as UInt64)
+
+        let account=self.account
+
+        let marketplaceReceiver=account.getCapability<&{FungibleToken.Receiver}>(/public/flowTokenReceiver)
+        let marketplaceNFTTrash: Capability<&{Art.CollectionPublic}> =account.getCapability<&{Art.CollectionPublic}>(Art.CollectionPublicPath)
 
-        self.account.save(<- create Administrator(), to: self.VersusAdministratorStoragePath)
-        self.account.link<&Administrator>(self.VersusAdministratorPrivatePath, target: self.VersusAdministratorStoragePath)
+        if !marketplaceNFTTrash.check() {
+            account.save<@NonFungibleToken.Collection>(<- Art.createEmptyCollection(), to: Art.CollectionStoragePath)
+            account.link<&{Art.CollectionPublic}>(Art.CollectionPublicPath, target: Art.CollectionStoragePath)
+        }
+
+        let versusCapability = account.getCapability<&{Versus.PublicDrop}>(Versus.CollectionPublicPath)
+        if !versusCapability.check() {
```

Comment:
There shouldn't be any case where `check()` returns true here, right? Since this is happening in the contract initializer. Do you even need the conditional?

Reply:
If you update the contract then the capability will already be stored.